lib/matplotlib/widgets.py

```diff
@@ -2529,26 +2591,36 @@ def extents(self, extents):
         self.set_visible(self.visible)
         self.update()
 
+    def _set_corner_width_rotation(self, xy0, width, height, rotation):
+        """
+        Set corner coordinate, width, height, and rotation.
+        """
+        xmin = xy0[0]
+        ymin = xy0[1]
+        xmax = xmin + width
+        ymax = ymin + height
+        if self.drawtype == 'box':
+            self.to_draw.set_angle(np.rad2deg(rotation))
+        self.extents = (xmin, xmax, ymin, ymax)
+
     def draw_shape(self, extents):
         x0, x1, y0, y1 = extents
-        xmin, xmax = sorted([x0, x1])
-        ymin, ymax = sorted([y0, y1])
         xlim = sorted(self.ax.get_xlim())
         ylim = sorted(self.ax.get_ylim())
 
-        xmin = max(xlim[0], xmin)
-        ymin = max(ylim[0], ymin)
-        xmax = min(xmax, xlim[1])
-        ymax = min(ymax, ylim[1])
+        x0 = np.clip(x0, *xlim)
+        x1 = np.clip(x1, *xlim)
+        y0 = np.clip(y0, *ylim)
+        y1 = np.clip(y1, *ylim)
```

Comment:
This doesn't seem right, as a rotated rectangle can be wider than the x limits, but these extents are for the un-rotated rectangle..

Reply:
I'm not sure why these lines were here in the first place, so I've just removed this bit.